R/climdex.R

```diff
@@ -0,0 +1,126 @@
+#' climdex
+#'
+#' @description
+#'  This function calculates climate indices based on temperature and precipitation data.  
+#'  
+#' @param data A data frame containing the temperature and precipitation data.
+#' @param station The name of the column in the data frame that represents the station.
+#' @param date The name of the column in the data frame that represents the date.
+#' @param year The name of the column in the data frame that represents the year.
+#' @param month The name of the column in the data frame that represents the month.
+#' @param prec The name of the column in the data frame that represents precipitation.
+#' @param tmax The name of the column in the data frame that represents maximum temperature.
+#' @param tmin The name of the column in the data frame that represents minimum temperature.
+#' @param indices A character vector specifying the climate indices to be calculated.
+#' @param freq The frequency at which the indices should be calculated. Can be "annual" or "monthly". Defaults to "annual".
+#' @param base.range A numeric vector specifying the base range for calculating indices. Defaults to c(1961, 1990).
+#' @param n The number of years to be considered for calculating percentile-based indices. Defaults to 5.
+#' @param northern.hemisphere A logical value indicating whether the data is from the Northern Hemisphere. Defaults to TRUE.
+#' @param quantiles A numeric vector specifying the quantiles for calculating percentile-based indices. Defaults to NULL.
+#' @param temp.qtiles A numeric vector specifying the quantiles for temperature-based indices. Defaults to c(0.1, 0.9).
+#' @param prec.qtiles A numeric vector specifying the quantiles for precipitation-based indices. Defaults to c(0.95, 0.99).
+#' @param max.missing.days A numeric vector specifying the maximum number of missing days allowed for each frequency. Defaults to c(annual = 15, monthly = 3).
+#' @param min.base.data.fraction.present The minimum fraction of base data required for calculating indices. Defaults to 0.1.
+#' @param spells.can.span.years A logical value indicating whether climate spells can span across years. Defaults to FALSE.
+#' @param gsl.mode The method to calculate the growing season length. Can be "GSL" or "ASL". Defaults to "GSL".
+#' @param threshold The threshold for calculating wet and dry spell indices. Defaults to 1.
+#'
+#' @return A data frame containing the calculated climate indices.
+#' 
+#' @export
+#'
+#' @examples
+#' data <- read.csv("climate_data.csv")
+#' indices <- c("fd", "su", "id", "tr", "wsdi", "csdi", "gsl", "sdii", "r10mm", "r20mm", "rnnmm", "cdd", "cwd", "r95ptot", "r99ptot", "prcptot")
+#' climdex(data, station = "station_name", date = "date", year = "year", month = "month", prec = "precipitation",
+#'          tmax = "max_temperature", tmin = "min_temperature", indices = indices)
```

Comment:
Nice example, but because this csv file doesn't exist can you comment out the examples?
(so run another "#" before the code - i.e.:

```
#' # data <- read.csv("climate_data.csv")
#' # indices <- c("fd", "su", "id", "tr", "wsdi", "csdi", "gsl", "sdii", "r10mm", "r20mm", "rnnmm", "cdd", "cwd", "r95ptot", "r99ptot", "prcptot")
#' # climdex(data, station = "station_name", date = "date", year = "year", month = "month", prec = "precipitation",
#' #          tmax = "max_temperature", tmin = "min_temperature", indices = indices)
```

Reply:
I can make these changes if it is easier for you because of branches - I only haven't made them through the GitHub editor because we can't run `devtools::document()` on the GitHub online editor